source/extensions/filters/http/original_src/original_src.cc

```diff
@@ -0,0 +1,50 @@
+#include "extensions/filters/http/original_src/original_src.h"
+
+#include "common/common/assert.h"
+
+#include "extensions/filters/common/original_src/socket_option_factory.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace OriginalSrc {
+
+OriginalSrcFilter::OriginalSrcFilter(const Config& config) : config_(config) {}
+
+void OriginalSrcFilter::onDestroy() {}
+
+Http::FilterHeadersStatus OriginalSrcFilter::decodeHeaders(Http::HeaderMap&, bool) {
+  const auto downstream_address = callbacks_->streamInfo().downstreamRemoteAddress();
+  ASSERT(downstream_address);
+
+  ENVOY_LOG(debug,
+            "Got a new connection in the original_src filter for address {}. Marking with {}",
+            downstream_address->asString(), config_.mark());
+
+  if (downstream_address->type() != Network::Address::Type::Ip) {
```

Comment:
Maybe this should come before the ENVOY_LOG call?

Reply:
Makes sense. It's misleading otherwise.